packages/vue/src/components/sign-up.vue

```diff
@@ -24,18 +24,21 @@ const onInput = (e: Event): void => {
 
   if (type === 'checkbox' && !checked)
     (value as string | undefined) = undefined;
+
   props.updateForm({ name, value });
 };
 const onSignUpSubmit = (e: Event): void => {
   if (attrs?.onSignUpSubmit) {
     emit('signUpSubmit', e);
   } else {
-    submit();
+    submit(e);
   }
 };
 
-const submit = (): void => {
-  props.submitForm();
+const submit = (e: Event): void => {
+  const formData = new FormData(<HTMLFormElement>e.target);
+  const json = Object.fromEntries(formData);
+  props.submitForm(json);
```

Comment:
Now it's submitting the json info. Not sure if handleInput is needed or not

Reply:
`handleInput` is still necessary for dynamically computing the `username` from the `country_code` + `phone` and persisting it in the form for submission.